pkg/api/util/db_test.go

```diff
@@ -682,3 +682,160 @@ func TestDeleteFileShareAclDBEntry(t *testing.T) {
+		in.ProfileId, in.Type = "d2975ebe-d82c-430f-b28e-f373746a71ca", "system"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid fileshare type: %v. Supported type is: ip", in.Type)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Empty Access To", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", ""
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := "accessTo is empty. Please give valid ip segment"
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip Segment", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.8/9.9"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid IP segment %v", in.AccessTo)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("%v is not a valid ip. Please give the proper ip", in.AccessTo)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Empty accesscapability", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo, in.AccessCapability = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.9", []string{}
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
```

Comment:
here, we are sending invalid profile, invalid type, invalid AccessTo and invalid AccessCapability. How are we guaranteed to receive only the CapabilityError in this case ? and AccessTo error in earlier case ? Something is amiss here.

Reply:
How are you deciding those profile are invalid, Could you please tell me how are you deciding it is invalid? Then i can anser your question